layer1/transaction/tests/transaction_test.go

```diff
@@ -0,0 +1,264 @@
+//go:build integration
+
+package tests
+
+import (
+	"context"
+	"github.com/alicenet/alicenet/layer1/ethereum"
+	"github.com/alicenet/alicenet/layer1/tests"
+	"github.com/alicenet/alicenet/test/mocks"
+	"github.com/ethereum/go-ethereum/core/types"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/alicenet/alicenet/layer1/transaction"
+	"github.com/stretchr/testify/assert"
+)
+
+func Setup(t *testing.T, accounts int, pollingTime time.Duration) (*tests.ClientFixture, *transaction.FrontWatcher) {
+	fixture := setupEthereum(t, accounts)
+	db := mocks.NewTestDB()
+	watcher := transaction.WatcherFromNetwork(fixture.Client, db, false, pollingTime)
+
+	return fixture, watcher
+}
+
+func TestSubscribeAndWaitForValidTx(t *testing.T) {
+	numAccounts := 2
+	fixture, watcher := Setup(t, numAccounts, 1*time.Second)
+	eth := fixture.Client
+	accounts := eth.GetKnownAccounts()
+	assert.Equal(t, numAccounts, len(accounts))
+
+	owner := accounts[0]
+	user := accounts[1]
+
+	ctx, cf := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cf()
+
+	amount := big.NewInt(12_345)
+	txn, err := ethereum.TransferEther(eth, fixture.Logger, user.Address, owner.Address, amount)
+	assert.Nil(t, err)
+
+	receipt, err := watcher.SubscribeAndWait(ctx, txn, nil)
+	assert.Nil(t, err)
+	assert.NotNil(t, receipt)
+	assert.Equal(t, txn.Hash(), receipt.TxHash)
+
+	currentHeight, err := eth.GetCurrentHeight(ctx)
+	assert.Nil(t, err)
+	assert.GreaterOrEqual(t, currentHeight, receipt.BlockNumber.Uint64()+eth.GetFinalityDelay())
+
+	_, isPending, err := eth.GetTransactionByHash(ctx, txn.Hash())
+	assert.Nil(t, err)
+	assert.False(t, isPending)
+
+	mintTxnOpts, err := eth.GetTransactionOpts(ctx, user)
+	assert.Nil(t, err)
+	mintTxnOpts.NoSend = false
+	//mintTxnOpts.GasFeeCap = big.NewInt(1_000_000_000)
+	mintTxnOpts.Value = amount
+
+	mintTxn, err := fixture.Contracts.EthereumContracts().BToken().MintTo(mintTxnOpts, owner.Address, big.NewInt(1))
+	assert.Nil(t, err)
+	assert.NotNil(t, mintTxn)
+
+	txnRough := &types.DynamicFeeTx{}
```

Comment:
What exactly are we testing from here onwards? Don't see any asserts and nothing is sent to the watcher o.o